jsonrpc/dispatcher.go

```diff
@@ -113,6 +114,26 @@ type wsConn interface {
 	WriteMessage(messageType int, data []byte) error
 }
 
+
+// as per https://www.jsonrpc.org/specification, the `id` in JSON-RPC 2.0 
+// can only be a string or a non-decimal integer
+func formatFilterResponse(id interface{}, resp string) (string, error) {
+	switch t := id.(type) {
+	case string:
+		return fmt.Sprintf(`{"jsonrpc":"2.0","id":%s,"result":"%s"}`, t, resp), nil
+	case float64:
+		if t == math.Trunc(t) {
+			return fmt.Sprintf(`{"jsonrpc":"2.0","id":%d,"result":"%s"}`, int(t), resp), nil
+		} else {
+			return "", invalidJSONRequest
+		}
+	case nil:
+		return fmt.Sprintf(`{"jsonrpc":"2.0","result":"%s"}`, resp), nil
```

Comment:
I think that by the [JSON-RPC spec](https://www.jsonrpc.org/specification) `id` is required in the `response object`, at least as `Nil`:
<img width="669" alt="Screen Shot 2021-08-05 at 16 16 21 PM" src="https://user-images.githubusercontent.com/10708079/128365600-488e6065-c17b-4413-97ed-d3ce5fabafb0.png">

Reply:
Nice catch will fix

Reply:
Okay added that